### ISSUE TITLE
The one-shot learning for face recognition #2

### ISSUE THREAD

**Issue by Han-sss** [Han-sss/Hello-world#2]
# The one-shot learning for face recognition
I rearranged the training set and validation set as the professor, you said. The sampled images in the validation set will not appear in the training set, but the corresponding persons will appear in the training set (and I also made sure that the images are sampled in the same video).

In this experiment, I kept the margin as a constant 2, and kept adjusting the number of unfreezing convolution layers and the update frequency of the learning rate.

### Uncropprd dataset
CONV4: False False False; CONV5: True, True, True
![cat(3)](https://user-images.githubusercontent.com/63163321/123621996-b6232900-d846-11eb-9dcc-527ef4badf5f.png)

CONV4: False False True; CONV5: True, True, True
![cat](https://user-images.githubusercontent.com/63163321/123614169-0bf3d300-d83f-11eb-85b4-abbc7bce0898.png)

CONV4: False True True; CONV5: True, True, True
![cat(1)](https://user-images.githubusercontent.com/63163321/123615739-8e30c700-d840-11eb-8261-6c7f65b066c5.png)

CONV4: True True True; CONV5: True, True, True
![cat(2)](https://user-images.githubusercontent.com/63163321/123619482-2e3c1f80-d844-11eb-99a5-16e4ec60495e.png)

### Cropped dataset
CONV4: False False False; CONV5: False, False, False
![cat(10)](https://user-images.githubusercontent.com/63163321/123632439-e40e6a80-d852-11eb-9059-89bd9aa015f0.png)

CONV4: False False False; CONV5: False, False, True
![cat(9)](https://user-images.githubusercontent.com/63163321/123630964-22a32580-d851-11eb-914d-69a882ecf174.png)

CONV4: False False False; CONV5: False, True, True
![cat(8)](https://user-images.githubusercontent.com/63163321/123630189-326e3a00-d850-11eb-81e4-cc73296f8dfa.png)

CONV4: False False False; CONV5: True, True, True
![cat(4)](https://user-images.githubusercontent.com/63163321/123627348-dbb33100-d84c-11eb-9b8b-a87f79c61775.png)

CONV4: False False True; CONV5: True, True, True
![cat(5)](https://user-images.githubusercontent.com/63163321/123628050-a78c4000-d84d-11eb-8b85-c9dae4bab1da.png)

CONV4: False True True; CONV5: True, True, True
![cat(6)](https://user-images.githubusercontent.com/63163321/123628769-6f393180-d84e-11eb-9824-339435f7fbbf.png)

CONV4: True True True; CONV5: True, True, True
![cat(7)](https://user-images.githubusercontent.com/63163321/123629416-3baad700-d84f-11eb-8893-e5f76ac0d6ac.png)

## Summary
As shown in the figure, while the model discriminates different person more accurately, the model also does not easily recognize the same person. I tried to adjust the value of margin in ContrastiveLoss, and the effect approximates to freeing more convolution layers but performs a bit worse.
